Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -627,20 +631,41 @@ open class ComposerVC: _ViewController,
     ///   - typingMention: The potential user mention the current user is typing.
     ///   - mentionRange: The position where the current user is typing a mention to it can be replaced with the suggestion.
     open func showMentionSuggestions(for typingMention: String, mentionRange: NSRange) {
-        userSearchController.search(
-            query: queryForMentionSuggestionsSearch(typingMention: typingMention)
-        )
+        guard let channel = channelController?.channel else {
+            return
+        }
+        guard let currentUserId = channelController?.client.currentUserId else {
+            return
+        }
+        
+        var usersCache: [ChatUser] = []
+
+        if mentionAllAppUsers {
+            userSearchController.search(
+                query: queryForMentionSuggestionsSearch(typingMention: typingMention)
+            )
+        } else {
+            usersCache = searchUsers(
+                channel.watchers.map { $0 } + channel.cachedMembers.map { $0 },
+                by: typingMention,
+                excludingId: currentUserId
+            )
+        }
+
         let dataSource = ChatMessageComposerSuggestionsMentionDataSource(
             collectionView: suggestionsVC.collectionView,
-            searchController: userSearchController
+            searchController: userSearchController,
+            usersCache: usersCache
         )
         suggestionsVC.dataSource = dataSource
         suggestionsVC.didSelectItemAt = { [weak self] userIndex in
             guard let self = self else { return }
+            guard dataSource.usersCache.count >= userIndex else {
+                return
+            }
 
             let textView = self.composerView.inputMessageView.textView
-            let user = self.userSearchController.users[userIndex]
-
+            let user = dataSource.usersCache[userIndex]
```

Comment:
What is the data comes from remote? This would not work right? Since it assumes we are always using the usersCache?